src/tir/transforms/lower_cross_thread_reduction.cc

```diff
@@ -155,6 +161,277 @@ Array<Buffer> MakeScratchpads(const Array<Buffer>& reduction_buffers, bool is_cr
   return new_buffers;
 }
 
+/*!
+ * \brief Get init value from BufferStore Node
+ * \param block The block to be checked
+ * \return The init value
+*/
+class InitUpdateValueFinder : public StmtExprVisitor {
+ public:
+  /*!
+   * \brief Find the init value of the given block
+   * \param block The block to be checked
+   * \return The init value of the given block
+   */
+  static PrimExpr FindInit(const Block& block) {
```

Comment:
by the usage in context, maybe we could merge the `FindInit()` and `CheckHasMMA()` to visit the block only once.